### PR TITLE
CXP-1288: Added mapping tab to catalog

### PR DESCRIPTION
**Pull request by RiffVentura**
mapping-layout2

### CODE REVIEW

File: components/catalogs/front/src/components/ProductMapping/ProductMapping.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    display: flex;
+    gap: 40px;
+`;
+const TargetContainer = styled.div`
+    flex-basis: 50%;
+    flex-grow: 1;
+`;
+const SourceContainer = styled.div`
+    flex-basis: 50%;
+    animation: ${slideIn} 0.2s ease-in-out;
```
Comment (RiffVentura) [inline components/catalogs/front/src/components/ProductMapping/ProductMapping.tsx:27]: added jokingly at first, then wondered if I should keep it
Easy to remove

Comment (tseho) [reply components/catalogs/front/src/components/ProductMapping/ProductMapping.tsx:27]: to be validated with Sophie :)

File: components/catalogs/front/src/components/CatalogEdit/CatalogEdit.integration.tsx [hunk 2]
```diff
@@ -21,6 +21,7 @@ const openDropdown = (selector: string): void => {
 };
 
 test('it can enable a catalog', () => {
+    const catalogId = 'catalog_id';
```
Comment (tseho) [inline components/catalogs/front/src/components/CatalogEdit/CatalogEdit.integration.tsx:24]:
```suggestion
```

No need for a variable if you don't use it.

File: components/catalogs/front/src/components/CatalogEdit/CatalogEdit.integration.tsx [hunk 3]
```diff
@@ -36,7 +37,7 @@ test('it can enable a catalog', () => {
         <ThemeProvider theme={pimTheme}>
             <QueryClientProvider client={new QueryClient()}>
                 <CatalogFormContext.Provider value={dispatch}>
-                    <CatalogEdit form={form} />
+                    <CatalogEdit id={catalogId} form={form} />
```
Comment (tseho) [inline components/catalogs/front/src/components/CatalogEdit/CatalogEdit.integration.tsx:40]:
```suggestion
                    <CatalogEdit id="8347c371-22c8-4642-aa26-b58a24f546b8" form={form} />
```

File: components/catalogs/front/src/components/CatalogEdit/CatalogEdit.integration.tsx [hunk 4]
```diff
@@ -50,6 +51,7 @@ test('it can enable a catalog', () => {
 test('it can change criteria in the product selection', async () => {
     mocked(generateRandomId).mockReturnValue('rdn');
 
+    const catalogId = 'catalog_id';
```
Comment (tseho) [inline components/catalogs/front/src/components/CatalogEdit/CatalogEdit.integration.tsx:54]:
```suggestion
```

File: components/catalogs/front/src/components/CatalogEdit/CatalogEdit.integration.tsx [hunk 5]
```diff
@@ -65,7 +67,7 @@ test('it can change criteria in the product selection', async () => {
         <ThemeProvider theme={pimTheme}>
             <QueryClientProvider client={new QueryClient()}>
                 <CatalogFormContext.Provider value={dispatch}>
-                    <CatalogEdit form={form} />
+                    <CatalogEdit id={catalogId} form={form} />
```
Comment (tseho) [inline components/catalogs/front/src/components/CatalogEdit/CatalogEdit.integration.tsx:70]:
```suggestion
                    <CatalogEdit id="8347c371-22c8-4642-aa26-b58a24f546b8" form={form} />
```

File: components/catalogs/front/src/components/CatalogEdit/CatalogEdit.integration.tsx [hunk 6]
```diff
@@ -99,6 +101,7 @@ test('it can add a product value filter on the channel', async () => {
         },
     ]);
 
+    const catalogId = 'catalog_id';
```
Comment (tseho) [inline components/catalogs/front/src/components/CatalogEdit/CatalogEdit.integration.tsx:104]:
```suggestion
```

File: components/catalogs/front/src/components/CatalogEdit/CatalogEdit.integration.tsx [hunk 7]
```diff
@@ -114,7 +117,7 @@ test('it can add a product value filter on the channel', async () => {
         <ThemeProvider theme={pimTheme}>
             <QueryClientProvider client={new QueryClient()}>
                 <CatalogFormContext.Provider value={dispatch}>
-                    <CatalogEdit form={form} />
+                    <CatalogEdit id={catalogId} form={form} />
```
Comment (tseho) [inline components/catalogs/front/src/components/CatalogEdit/CatalogEdit.integration.tsx:120]:
```suggestion
                    <CatalogEdit id="8347c371-22c8-4642-aa26-b58a24f546b8" form={form} />
```

File: components/catalogs/front/src/hooks/useCatalogMappingRequirements.ts [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+type Data = null;
+type Error = string | null;
+type Result = {
+    isLoading: boolean;
+    isError: boolean;
+    data: Data | undefined;
+    error: Error;
+};
+
+export const useCatalogMappingRequirements = (catalogId: string): Result => {
```
Comment (tseho) [inline components/catalogs/front/src/hooks/useCatalogMappingRequirements.ts:12]:
```suggestion
export const useProductMappingSchema = (catalogId: string): Result => {
```
WDYT ?

**Review by tseho — APPROVED** [review]

**Review by masacc — APPROVED** [review]

**Pull request merged**